src/components/dynamic-lists/dynamic-list.component.ts

```diff
@@ -88,11 +90,36 @@ export class BeagleListViewComponent
     })
   }
 
+  getColumnsQuantityStyle() {
+    return `repeat(${this.numColumns}, auto)`
+  }
+
+  getClassForType() {
+    return this.type === 'GRID' ? 'beagle-grid-view' :
+      `beagle-list-view ${this.direction}`
+  }
+
+  getRowCount() {
+    if (this.type === 'LIST') 
+      {return this.direction === 'VERTICAL' ? this.dataSource.length : 1}
```

Comment:
Use the suggestion of creating a String Enum for Type and Direction.

```suggestion
      {return this.direction === ListDirection.Vertical ? this.dataSource.length : 1}
```